src/query/storage/index.go

```diff
@@ -136,26 +163,39 @@ func queryKey(m models.Matchers) []byte {
 	return key
 }
 
-// FetchQueryToM3Query converts an m3coordinator fetch query to an M3 query
-func FetchQueryToM3Query(fetchQuery *FetchQuery, cache *QueryConversionCache) (index.Query, error) {
+// FetchQueryToM3Query converts an m3coordinator fetch query to an M3 query.
+func FetchQueryToM3Query(
+	fetchQuery *FetchQuery,
+	cache *QueryConversionCache,
+) (index.Query, error) {
 	matchers := fetchQuery.TagMatchers
-	k := queryKey(matchers)
+	// If no matchers provided, explicitly set this to an AllQuery
+	if len(matchers) == 0 {
+		return index.Query{
+			// TODO: change this to an idx.AllQuery: https://github.com/m3db/m3/pull/1478
```

Comment:
This returns every id if no matchers are provided? That seems not safe.

Reply:
To expound:

If not providing any queries, we want to return the set of all possible tags, and all values that each of those tags can have. We'll be building functionality for this in the index which should actually end up more performant than the new tag completion endpoint (perhaps counterintuitively, but we can get huge gains here when we're not required to filter down to satisfy an arbitrary query)

Reply:
Makes sense